substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JfrStackTraceElement.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jdk;
+
+public class JfrStackTraceElement {
+    private final StackTraceElement stackTraceElement;
```

Comment:
This is very expensive. Can we use a smaller representation?

Reply:
Good point! With the last [commit](https://github.com/oracle/graal/pull/3720/commits/0f8c6008777ac575b8afa53bbffb5ebc4623d672), `JfrStackTraceElement` contains only neccessary elements i.e. without reference to "fat" `java.lang.StackTraceElement`.